Add KimTrack to the Kimai Store

JAWS-tm opened the pull request:
Hi,
This PR adds KimTrack, a native menu-bar time tracker for Kimai (Windows, macOS & Linux), to the Kimai Store. 

Your feedback and any suggested changes are welcome!
>  The purchase link is currently in test mode (use card 4242 4242 4242 4242), I’ll switch it to the live URL once everything is approved. 

Thanks !

kevinpapst commented on _includes/store/playmoweb-kimtrack.md:
```diff
@@ -0,0 +1,82 @@
+A: Yes! KimTrack uses secure API calls to sync your data with your Kimai instance. Your information is safe and private on your device.
+
+
+<!-- ### App Screens
+
+<p>
+    <img src="/images/marketplace/ticesoftware-kimaibar-macos-client1.webp" width="45%" style="margin:0 10px">
+    <img src="/images/marketplace/ticesoftware-kimaibar-macos-client2.webp" width="45%" style="margin:0 10px">
+</p>    
+<p>
+    <img src="/images/marketplace/ticesoftware-kimaibar-macos-client3.webp" width="45%" style="margin:0 10px">
+    <img src="/images/marketplace/ticesoftware-kimaibar-macos-client4.webp" width="45%" style="margin:0 10px">
+</p> -->
+
+
+
+<!-- KimaiTrack is the time tracking solution running in your menu bar.
+
+Check your running time and quickly start and stop your timer directly from your menu bar. Switch to other projects without hassle with just another click on one of your recent activities. KimaiBar detects when you are idle for a certain time and asks you, what to do with the time when you come back. -->
```
Remove content or comment

kevinpapst commented on _includes/store/playmoweb-kimtrack.md:
```diff
@@ -0,0 +1,82 @@
+4. **Customize & Go**  
+   Head to **Settings** to choose your theme, set idle rules, and tailor notifications.
+
+---
+<p></p>
```
please remove

JAWS-tm replied:
I added those <p> tags because, without them, the separator sticks to the next title.

kevinpapst replied:
Weird. 
Anyway, the design (even if you don't like it) is consistent without separator 😁

kevinpapst commented on _includes/store/playmoweb-kimtrack.md:
```diff
@@ -0,0 +1,82 @@
+
+- **Customizable Preferences**  
+  Tailor notifications, idle rules, and display options. Your time-tracking, your way.
+
+
+---
+<p></p>
```
please remove

kevinpapst commented on _data/store/playmoweb-kimtrack.yml:
```diff
@@ -0,0 +1,14 @@
+developer: playmoweb
+icon: fas fa-desktop
+price: 10 €
```
Test mode says 15€, which one is correct?

JAWS-tm replied:
The correct is 10€ it will be the correct price in the live payment page

kevinpapst reviewed (changes requested):
Please add one Mac release to your repo, so I can test.
 
This is the Email I received, you might want to check the button font color:
<img width="595" alt="Bildschirmfoto 2025-05-02 um 17 37 07" src="https://github.com/user-attachments/assets/1ef400c7-75c8-4cfe-a40e-ec218305aca3" />

JAWS-tm commented:
I published the first release so you can try, I will add the missing Linux installer on Monday.

kevinpapst commented:
Thanks! 

Minor feedback: Font of tags in dark mode is unreadable.
<img width="162" alt="Bildschirmfoto 2025-05-02 um 17 57 28" src="https://github.com/user-attachments/assets/eb92c14e-6552-4822-9263-73d257cd4bda" />

An important problem: If I leave the window open and focus back to my working window (e.g. browser fullscreen) I can't easily get back to the KimTrack window. It does not show up in the tab switcher and the dropdown does not offer to bring the window back to the front. Maybe you can bring it to the front upon opening the menu:
<img width="122" alt="Bildschirmfoto 2025-05-02 um 18 03 56" src="https://github.com/user-attachments/assets/71a0cf0b-d5e2-4815-a3a4-5107b3fd80f2" />

JAWS-tm commented:
Hey,
Thanks for your detailed feedback – I’ve addressed all of the points you raised. 
Regarding the problem with the buttons you mentioned (including the timer on the right): they’re supposed to open the “Add Timesheet” modal. Are they not working on your end?

kevinpapst commented:
> Are they not working on your end?

No, that's why I mentioned it.

But now I can't even bring the main window back, no matter what I do. 
I even minimized all 20 app windows, but it is just gone. 
Even after restarting the app.
Please fix that (add a new menu entry?) and let me know when I can update and re-test.
This is a M3 Max, dual screen, running Sequoia 15.3.1

JAWS-tm commented:
Hey, sorry for the delay.
I've updated the app can you try again :
https://github.com/playmoweb/KimTrack/releases/download/v1.0.0/KimTrack-1.0.0.dmg

Thanks

kevinpapst commented:
Same problem: I am not able to see the main window or do anything except opening the License or Settings screen.
Unusable for me. 
The offer for the screenshare is still open.

JAWS-tm commented:
Hmm, that’s strange. Is your API configuration running? Try saving it in the settings to see if it can connect.

JAWS-tm commented:
Could you please let me know which version of your Kimai instance you used to test the app?

kevinpapst commented:
> Hmm, that’s strange. Is your API configuration running? Try saving it in the settings to see if it can connect.

Ahh, I think this was the issue: I used the demo for testing and it was reinstalled, so the API key was reset. 

Works now(!), but take that as a first feedback: if the app detects that some API call fails, you should display an error message and not only hide the main window silently.

The buttons do work as well (maybe the same issue?).

kevinpapst commented on _includes/store/playmoweb-kimtrack.md:
```diff
@@ -0,0 +1,57 @@
+- **Native Desktop App**  
+  Enjoy a lightweight, responsive interface on **macOS**, **Windows**, or **Linux**—no browser tabs required. Access all your time-tracking features directly from your menu bar for a seamless experience.
+
+- **Seamless Kimai API Integration**  
```
Users do not care about API integrations. They want their data to be up-to-date all the time. 
```suggestion
- **Seamless Kimai Sync**  
```

kevinpapst commented on _includes/store/playmoweb-kimtrack.md:
```diff
@@ -0,0 +1,57 @@
+2. **Activate Your License**  
+   - Open KimTrack and click **“Browse your license”**  
+   - Select the license file you received via email  
+   *(Internet connection required)*
```
>    *(Internet connection required)*

This should probably be mentioned in the first introduction.

kevinpapst commented on _data/en/store.yml:
```diff
@@ -137,6 +137,9 @@ items:
         intro: Time tracking for Mac, seamlessly integrated with Kimai – both On-Premise and Cloud.
+    playmoweb-kimtrack:
+        title: KimTrack
+        intro: Native desktop client for macOS, Windows, and Linux to track time from your menu bar.
```
```suggestion
        intro: Native desktop client for macOS, Windows, and Linux to track time from your menu bar.
```
Is your app really "native" ? It feels like a  Native desktop client is partly misleading here (e.g. its not Swift on MacOS)

BTW: there is no About dialog. 
Please include one and clarify that KimTrack is NOT affiliated with me. 
And include support information, I get many support requests for third party plugins/apps.

kevinpapst reviewed (changes requested):
Please check my comments and resolve the merge conflict.

JAWS-tm commented:
Hey,

Thanks for your feedback! I’ve updated the app to display when the API configuration becomes invalid and added an indicator in the API config tab. 
I’ve also revised the store information based on your suggestions and added the About window to the app.

Thanks again!

kevinpapst commented:
<img width="302" alt="Bildschirmfoto 2025-05-16 um 14 07 04" src="https://github.com/user-attachments/assets/7f2e32b3-ed24-47d8-b4de-113e43d46cb5" />

Perfect 👍

kevinpapst commented on _data/developer.yml:
```diff
@@ -179,3 +179,16 @@ charliemonroe:
+        - title: Where can I buy this plugin?
+          content: You can buy the plugin via the Marketplace 'Stripe' and pay with Credit Card.
+        - title: Where can I get support for this plugin?
+          content: Support-questions, bug-reports and feature-requests can be made via the <a href="mailto:contact@playmoweb.com">email address</a>.
```
```suggestion
          content: Support-questions, bug-reports and feature-requests can be made via the <a href="mailto:contact@playmoweb.com">contact@playmoweb.com</a>.
```

kevinpapst commented on _includes/store/playmoweb-kimtrack.md:
```diff
@@ -0,0 +1,58 @@
+   - Select the license file you received via email  
+
+3. **Connect to Kimai**  
+   - Enter your **API URL** (e.g. `https://your-kimai.example.com/`)  
```
```suggestion
   - Enter your **API URL** (e.g. `https://foobar.kimai.cloud/api/`)  
```

kevinpapst commented on _includes/store/playmoweb-kimtrack.md:
```diff
@@ -0,0 +1,58 @@
+
+3. **Connect to Kimai**  
+   - Enter your **API URL** (e.g. `https://your-kimai.example.com/`)  
+   - Paste your **Bearer Token** (found under **Profile → API Access** in Kimai)
```
```suggestion
   - Paste your **API Token** (create one in your **Profile → API Access** in Kimai)
```

kevinpapst commented on _includes/store/playmoweb-kimtrack.md:
```diff
@@ -0,0 +1,58 @@
+A: Our support team is just an email away at [contact@playmoweb.com](mailto:contact@playmoweb.com) or open an issue on [GitHub](https://github.com/playmoweb/kimtrack/issues).
+
+**Q: Can I use KimTrack with any Kimai instance?**  
+A: Absolutely! Just enter your API URL and Bearer Token, and you’re good to go.
```
```suggestion
A: Absolutely! Both Kimai Cloud and On-Premise installations are supported. Just enter your API URL and API Token, and you’re good to go.
```

kevinpapst commented on _data/store/playmoweb-kimtrack.yml:
```diff
@@ -0,0 +1,14 @@
+developer: playmoweb
+icon: fas fa-desktop
+price: 10 €
+shop: "https://buy.stripe.com/test_cN2fZWdrh2m4eB2aEE"
```
Please let me know when you fixed th URL

JAWS-tm replied:
That's should be good ;)

kevinpapst reviewed (changes requested):
LGTM 👍 

Will merge, once the payment URL is changed.

kevinpapst commented:
Works. The payment flow could be improved.

This is the final payment screen:
<img width="1009" alt="Bildschirmfoto 2025-05-16 um 14 52 46" src="https://github.com/user-attachments/assets/ee0c2e77-6198-429f-a7a1-df1112d50cfd" />

And then it took several minutes for the email to show up.
Can't you include a "back link" to the payment screen?
I believe this should be possible, including the payment ID, so you could show a Thank you page with an exaplanation about the next steps.

Anyway: how can I remove the registered test license and replace it with the one I purchased?

kevinpapst commented:
And two more: 
1. I did not receive an invoice. How can I get one? Don't you use Stripe billing?
2. The download link in the email could/should be improved. If you don't want to build a download page, at least point the link to the "releases" page. We know GitHub, but most users never saw the platform before.
<img width="646" alt="Bildschirmfoto 2025-05-16 um 15 08 22" src="https://github.com/user-attachments/assets/7b4fbe66-7fb9-4372-b00d-9f416c1baad8" />

JAWS-tm commented:
The email delay is normal, your payment was made just before our server restarted. I manually triggered the webhook afterward to complete the process.

We'll also look into improving the post-purchase redirect, likely by sending users directly to the GitHub page where setup instructions. In the future we will probably setup a dedicated download page.

Thanks for your purchase, for the invoice we are looking to enable correctly the generation, we will send it to you ASAP.

JAWS-tm commented:
To replace the license we haven't provided a way to do it from the UI right now. But you can go into the file located at `~/.kimai-prefs.json` and remove the `licenseId` property

kevinpapst commented:
Received an invoice now, thanks!
